src/pointerwrappers.jl

```diff
@@ -0,0 +1,43 @@
+module PointerWrappers
+
+# Pointer wrapper for conveniently accessing nested structures
+struct PointerWrapper{T}
+  pointer::Ptr{T}
+end
+
+# Non-pointer-type fields get wrapped as a normal PointerWrapper
+PointerWrapper(::Type{T}, pointer) where T = PointerWrapper{T}(pointer)
+
+# Pointer-type fields get dereferenced such that PointerWrapper wraps the pointer to the field type
+PointerWrapper(::Type{Ptr{T}}, pointer) where T = PointerWrapper{T}(unsafe_load(Ptr{Ptr{T}}(pointer)))
+
+# Cannot use `pw.pointer` since we implement `getproperty` to return the fields of `T` itself
+Base.pointer(pw::PointerWrapper{T}) where T = getfield(pw, :pointer)
+
```

Comment:
```suggestion
# Allow passing a `PointerWrapper` to wrapped C functions
Base.unsafe_convert(::Type{Ptr{T}}, pw::PointerWrapper{T}) where {T} = Base.unsafe_convert(Ptr{T}, pointer(pw))

```
?

Reply:
I am never sure: should we implement https://docs.julialang.org/en/v1/base/c/#Base.unsafe_convert or https://docs.julialang.org/en/v1/base/c/#Base.cconvert? Do you know the correct guidelines for this? The docs are as usual confusing (to me).

Reply:
The docstring of `cconvert` states

> Neither `convert` nor `cconvert` should take a Julia object and turn it into a `Ptr`.

Since that's what we are doing here, `unsafe_convert` seem to be okay, I think